### PR TITLE
feat(enterprise): returns supported instance types and capacity info in GET clusters by id endpoint

### PR DESCRIPTION
**Pull request by machi1990**
## Description

Returns the capacity information and supported instance types on the `GET /api/kafkas_mgmt/v1/clusters/{id}` endpoint The information is needed by API consumers when creating a Kafka on a dedicated cluster. Along this, cloud_provider and region information are returned as well.

NOTE: enterprise clusters supports only standard instances and `enterprise` billing model at the moment. The returned info takes this into account.

Closes https://issues.redhat.com/browse/MGDSTRM-10700

## Verification Steps

Added integration test should pass

## Checklist (Definition of Done)

- [x] All acceptance criteria specified in JIRA have been completed
- [x] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [x] Documentation added for the feature
- [x] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- ~~[ ] Required metrics/dashboards/alerts have been added (or PR created).~~
- ~~[ ] Required Standard Operating Procedure (SOP) is added.~~
- ~~[ ] JIRA has been created for changes required on the client side~~

### CODE REVIEW

File: openapi/kas-fleet-manager.yaml [hunk 1]
```diff
@@ -1513,7 +1514,49 @@ components:
...
+              description: "A flag indicating whether this cluster is multi availabability or not"
+              type: boolean
+    EnterpriseCluster:
+      allOf:
+        - $ref: "#/components/schemas/EnterpriseClusterListItem"
+        - type: object
+          properties:
+            supported_instance_types: 
+              type: object
+              $ref: "#/components/schemas/SupportedKafkaInstanceTypesList"
```
Comment (machi1990) [inline openapi/kas-fleet-manager.yaml:1533]: @riccardo-forina @dimakis this new field might interest you.

File: internal/kafka/internal/presenters/cluster.go [hunk 2]
```diff
@@ -33,14 +54,75 @@ func PresentEnterpriseClusterWithAddonParams(cluster api.Cluster, fleetShardPara
...
+		RemainingKafkaStreamingUnits: storedCapacityInfo.MaxUnits - consumedStreamingUnitsInTheCluster,
+		MaximumKafkaStreamingUnits:   storedCapacityInfo.MaxUnits,
+	}
+}
+
+func presentEnterpriseClusterSupportedInstanceTypes(remainingCapacity int32, kafkaConfig *config.KafkaConfig) public.SupportedKafkaInstanceTypesList {
+	// enterprise clusters only supports standard instance type for now. It is safe to hardcode this.
+	standardInstanceType, err := kafkaConfig.SupportedInstanceTypes.Configuration.GetKafkaInstanceTypeByID(types.STANDARD.String())
+	if err != nil { // this should never happen, lets log an error in case it happens.
+		logger.Logger.Errorf("failed to find standard instance type from supported instance type config due to %q.", err.Error())
```
Comment (pb82) [inline internal/kafka/internal/presenters/cluster.go:95]: are we generally not returning errors from presenter functions? This hides the error, although it should be quite rare here.

**Review by pb82 — APPROVED** [review]
LGTM

File: internal/kafka/internal/presenters/cluster.go (hunk 2 above)
Comment (machi1990) [reply internal/kafka/internal/presenters/cluster.go:95]: We could return error from the presenter.
My thinking here was - this error will never occur. If it occurs, then it is a configuration error. 
Let's log it to sentry and we can action it.

Comment (miguelsorianod) [reply internal/kafka/internal/presenters/cluster.go:95]: Both things can be performed

**Review by pawelpaszki — APPROVED** [review]
/lgtm

**Pull request merged**